tensor/objects.py

```diff
@@ -147,27 +162,104 @@ def __init__(self, config, queueBack, tensor):
         if self.hostname is None:
             self.hostname = socket.gethostbyaddr(socket.gethostname())[0]
 
-        self.tensor = tensor
+        self.use_ssh = config.get('use_ssh', False)
+
+        if self.use_ssh:
+            self._init_ssh()
 
         self.queueBack = self._queueBack(queueBack)
 
         self.running = False
 
+    def _init_ssh(self):
+        """ Configure SSH client options
+        """
+
+        self.ssh_host = self.config.get('ssh_host', self.hostname)
+
+        self.known_hosts = self.config.get('ssh_knownhosts_file',
+            self.tensor.config.get('ssh_knownhosts_file', None))
+
+        self.ssh_keyfile = self.config.get('ssh_keyfile',
+            self.tensor.config.get('ssh_keyfile', None))
+
+        self.ssh_key = self.config.get('ssh_key',
+            self.tensor.config.get('ssh_key', None))
+
+        # Not sure why you'd bother but maybe you've got a weird policy
+        self.ssh_keypass = self.config.get('ssh_keypass',
+            self.tensor.config.get('ssh_keypass', None))
+
+        self.ssh_user = self.config.get('ssh_username',
+            self.tensor.config.get('ssh_username', None))
+
+        self.ssh_password = self.config.get('ssh_password',
+            self.tensor.config.get('ssh_password', None))
+
+        self.ssh_port = self.config.get('ssh_port',
+            self.tensor.config.get('ssh_port', 22))
+
+        # Verify config to see if we're good to go
+
+        if not (self.ssh_key or self.ssh_keyfile or self.ssh_password):
+            raise Exception("To use SSH you must specify *one* of ssh_key,"
+                            " ssh_keyfile or ssh_password for this source"
+                            " check or globally")
+
+        if not self.ssh_user:
+            raise Exception("ssh_username must be set")
+
+        self.ssh_keydb = []
+
+        cHash = hashlib.sha1(('%s:%s:%s:%s:%s:%s' % (
+            self.ssh_host, self.ssh_user, self.ssh_port, self.ssh_password,
+            self.ssh_key, self.ssh_keyfile)).encode()).hexdigest()
```

Comment:
Quite a few less bits of memory in exchange for some upfront CPU usage was my thinking.